site/pages/pledge/[address].tsx

```diff
@@ -83,6 +96,62 @@ export const getStaticProps: GetStaticProps<PageProps, Params> = async (
       pledge = { ...DEFAULT_VALUES, ownerAddress: resolvedAddress };
     }
 
+    // add up retirements here
+    let retirements: RetirementsTotalsAndBalances;
+    if (pledge.wallets && Object.values(pledge.wallets).length) {
+      const verifiedWallets = Object.values(pledge.wallets).filter(
+        (wallet) => wallet.status === "verified"
+      );
+      const promises = verifiedWallets.map((wallet) =>
+        getRetirementTotalsAndBalances({
+          address: wallet.address,
+        })
+      );
+      promises.push(
+        getRetirementTotalsAndBalances({
+          address: resolvedAddress,
+        })
+      );
+      const values: RetirementsTotalsAndBalances[] = await Promise.all(
+        promises
+      );
+      values.reduce((prev: any, curr) => {
```

Comment:
this should return 1 reduced value at the end. We can also use a Generic type arg <ExpectedType> so it knows what to expect for the `prev` and `return` values

```ts
const sumValues = values.reduce<ExpectedType>((prev, curr) => {
   ...dostuff()
}, SumValuesType)
```